enhancements/network/egressip_for_ovn_kubernetes.md

```diff
@@ -0,0 +1,148 @@
+re-distributed to another node in the cluster which can host them.
+
+All in all, the feature is quite complex as it introduces a lot of notions which obfuscate these two final functionalities/goals.  
+
+This enhancement proposal focuses on how this feature will work with OVN-Kubernetes.
+
+## Motivation
+
+The primary motivation is making OVN-Kubernetes have feature parity with openshift-sdn. Secondary motivations come partially from 
+product management: enhancing the feature by allowing cluster admins the freedom to control cluster egress traffic 
+on pod level, and partially from us in the networking team; simplifying the usability for cluster admins and our 
+maintenance of the code base which implements it.   
+
+### Goals & Requirements
+
+2.1. The egress IP functionality for OVN-Kubernetes needs to have feature parity with openshift-sdn, according to 1.1. and 1.2.
+2.2. The egress IP functionality for OVN-Kubernetes should allow pod selection within a namespace when defining the IP of egress traffic. 
+2.3. The egress IP functionality for OVN-Kubernetes should support single-stack IPv4/IPv6 as well as dual-stack, in line with global 
+OpenShift requirements for all networking functionality for release 4.6.
+
+This proposal details how OVN-Kubernetes goes about addressing these goals in the section [Proposal](#proposal)
+ 
+### Non-Goals
+
+3.1. Making an EgressIP object cluster-wide, it will still be namespaced
+
+## Proposal
+
+We propose to make the assignment of egress IPs simpler by defining the following sequence of events for a user wishing to define an egress 
+IP for its cluster.
+
+As to satisfy 1.1. we propose that cluster admins tag the nodes they wish to use as egress nodes by labelling them with 
+`k8s.ovn.org/egress-assignable: ""`. In case an `EgressIP` object is created and no nodes are tagged as such, the egress IP creation is 
+invalidated. To rectify that: a cluster admin would have to; delete the previously created `EgressIP` object, 
+label the nodes needed, and re-create the previous object. 
+
+As to satisfy 2.2. we propose adding a `PodSelector.MatchLabels` field (following the convention for such a field for Kubernetes / OpenShift) 
+to the `EgressIP.Spec` 
+
+In summary: as to satisfy all goal/requirements we propose the following EgressIP object definition:
+
+```
+apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+  name: egressip
+  namespace: egressip-namespace
+spec:
+  egressIPs:
+  - 192.168.126.11
+  - 192.168.126.102
+  podSelector:
+    matchLabels:
+      name: my-pod
+status:
+  assignments:
+  - node: nodeX
+    egressIP: 192.168.126.11
+  - node: nodeY
+    egressIP: 192.168.126.102
+```
+
+In case `nodeX` or `nodeY` go down, the assigned IP is re-distributed on another node in the cluster that has the label defined above
+and is able to host those IPs.
+
+### Constraints
+
+4.1. No pod can match multiple `EgressIP` objects, leading to multiple egress IP assignments at the same time. 
```

Comment:
How is this enforced?

I would probably just say that if a pod matches multiple `EgressIP`s, it's undefined which one it ends up using.

You should also say explicitly whether this means that you can't have a selector-less `EgressIP` object in a namespace that also has selector-ful `EgressIP`s (since the pods that matched the selector-ful one would also match the selector-less one)

Reply:
I have not enforced it yet. And thinking about gives me headaches, so I think we should just not support it and default to "it's undefined" as you mentioned. I expand on the documentation